sql/analyzer/rules.go

```diff
@@ -809,3 +810,49 @@ func fixFieldIndexes(schema sql.Schema, exp sql.Expression) (sql.Expression, err
 		return e, nil
 	})
 }
+
+func fixJoinFields(ctx *sql.Context, a *Analyzer, node sql.Node) (sql.Node, error) {
```

Comment:
Can you add some comments to this function explaining why this is needed and adding a TODO to fix it in a proper way?